### PR TITLE
RavenDB-23341 removed single file app

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-23341

### Additional description

Reverts:
- https://github.com/ravendb/ravendb/pull/18517
- https://github.com/ravendb/ravendb/pull/19113
- https://github.com/ravendb/ravendb/pull/19119
- https://github.com/ravendb/ravendb/pull/19044
- https://github.com/ravendb/ravendb/pull/18756

Reasoning: Due to increased complexity in debugging experience caused by lack of proper tooling support (MS and 3rd party) and requirement to take only full memory dumps (other dump types are not supported at all - design decision for SFA), we decided to drop that feature because benefits do not outweigh the issues.

### Type of change

- [ ] Bug fix
- [x] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [ ] Low 
- [x] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [x] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [x] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [ ] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by hiddenshadow21 — APPROVED** [review]

File: build.ps1 [hunk 1]
```diff
@@ -207,12 +207,11 @@ Foreach ($target in $targets) {
...
-    BuildDummyApp $specOutDir $target
     BuildServer $SERVER_SRC_DIR $specOutDir $target
-    BuildTool rvn $RVN_SRC_DIR $specOutDir $target
-    BuildTool drtools $DRTOOL_SRC_DIR $specOutDir $target
-    BuildTool migrator $MIGRATOR_SRC_DIR $specOutDir $target
-    BuildDebug $DEBUG_SRC_DIR $specOutDir $target
+    BuildTool rvn $RVN_SRC_DIR $specOutDir $target $true
+    BuildTool drtools $DRTOOL_SRC_DIR $specOutDir $target $false
+    BuildTool migrator $MIGRATOR_SRC_DIR $specOutDir $target $true
+    BuildTool debug $DEBUG_SRC_DIR $specOutDir $target $true
```
Comment (gregolsky) [inline build.ps1:214]: the flag does not seem to be in use in `BuildTool()` - usage is commented out - maybe we can  remove these true/false

Comment (ppekrol) [reply build.ps1:214]: I copied it back 1:1 from 6.0 branch, didn't want to make any new additions

File: scripts/linux/pkg/deb/assets/build.sh [hunk 2]
```diff
@@ -108,7 +108,6 @@ done
 find /build -type d -exec chmod 755 {} +
 find /build -type f -exec chmod 644 {} +
 
-export DEB_BUILD_OPTIONS="nostrip"
```
Comment (gregolsky) [inline scripts/linux/pkg/deb/assets/build.sh:111]: not sure about this one

**Review by gregolsky — COMMENTED** [review]
Please see my comments

**Comment by gregolsky** [comment]
There was a piece somewhere about the DLLs hashes verification. Can you confirm that is intact?

**Comment by ppekrol** [comment]
its the `Copy-FileHash.ps1` - wasn't touched even when we moved to SFA

**Review by gregolsky — APPROVED** [review]

**Pull request merged**